flink-ml-iteration/src/main/java/org/apache/flink/ml/iteration/operator/OperatorWrapper.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.iteration.operator;
+
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.streaming.api.operators.StreamOperator;
+import org.apache.flink.streaming.api.operators.StreamOperatorFactory;
+import org.apache.flink.streaming.api.operators.StreamOperatorParameters;
+
+import java.io.Serializable;
+
+/** Wrappers for the given operator factory. */
+public interface OperatorWrapper<T, R> extends Serializable {
```

Comment:
Is this interface necessary? It looks that this should only have one implementation.

Reply:
There are two implements in the following PRs as a whole:
1. AllRoundOperatorWrapper: with this wrapper all the round is processed by a single operator instance.
2. PerRoundOperatorWrapper: with this wrapper each round of data is processed by a new operator instance. This enables us to reuse the operators defined outside of the iteration.